test/test_utils/ec2.py

```diff
@@ -38,7 +38,7 @@ def get_ec2_instance_type(default, processor, enable_p3dn=False):
         )
     instance_type = os.getenv(f"EC2_{processor.upper()}_INSTANCE_TYPE", default)
     if instance_type == p3dn and not enable_p3dn:
-        return [default]
+        instance_type = default
```

Comment:
is part of the scope of this PR to enable all the p3dn tests? if so, how about we make enable_p3dn=True by default in the function, since it seems the new default behavior should have p3dns enabled?

Reply:
I think this is better because we want to affirmatively enable p3dn instance usage through a PR rather than incidentally enabling p3dn instance usage through a random change.

Our focus seems to be more on preventing p3dn instances from being used unnecessarily than allowing the use of p3dn instances on all tests.

Reply:
the initial reason behind having enable_p3dn as an option was because p3dn testing was disabled by default due to resource issues. If this fixes the resource issues, I think we should no longer make p3dn an opt-in, to avoid mistakenly not enabling

Reply:
Given that we are now empowered to use p3dns, and your concerns about unnecessary p3dn usage, I think an implementation like the following might make sense:

1) Change the semantics to "disable_p3dn" -- this way we are communicating that p3dn testing is enabled by default
2) never allow p3dn to be the "default" instance type -- i.e., raise error if someone uses p3dn as the default instance type -- this will help avoid unnecessary p3dn usage in PR context.

Reply:
1 make sense.
2 seems unnecessary. We might have some tests where a p3dn instance is the best to use.